packages/e2e/helpers/dappwright.ts

```diff
@@ -1,80 +1,83 @@
+
+type IntRange<F extends number, T extends number> = Exclude<Enumerate<T>, Enumerate<F>> | T;
+
 let sharedBrowserContext: BrowserContext;
 
+// Fast mode limits the number of accounts created to 1 per role
+const fastMode = process.env.TEST_FAST_MODE === "true";
+
 export class RoleSwitcher {
   private wallet: Dappwright;
 
   constructor(wallet: Dappwright) {
     this.wallet = wallet;
   }
 
-  private async changeToMember(
-    indexes: number[],
-    memberNumber: number,
-    groupName: string,
-    page: Page
-  ) {
-    const selectedCouncil = indexes[memberNumber];
-    if (selectedCouncil === undefined) {
-      throw new Error(
-        `There is only ${indexes.length} ${groupName} members. Received: ${memberNumber}`
-      );
+  async council(councilNumber: IntRange<1, typeof COUNCIL_SIZE>, page?: Page) {
+    if (fastMode && councilNumber !== 1) {
+      throw new Error("Fast mode does not support council numbers other than 1");
     }
-    await this.switchToIndex(selectedCouncil, page);
-    await page.bringToFront();
-    await page.reload();
+    const index = COUNCIL_INDEXES[councilNumber - 1] as number;
+    await this.switchToIndex(index, page);
   }
 
-  async council(page: Page, councilNumber = 0): Promise<void> {
-    await this.changeToMember(ALL_COUNCIL_INDEXES, councilNumber, "security council", page);
+  async guardian(guardianNumber: IntRange<1, typeof GUARDIANS_SIZE>, page?: Page) {
+    if (fastMode && guardianNumber !== 1) {
+      throw new Error("Fast mode does not support guardian numbers other than 1");
+    }
+    const index = GUARDIAN_INDEXES[guardianNumber - 1] as number;
+    await this.switchToIndex(index, page);
   }
 
-  async guardian(page: Page, guardianNumber = 0): Promise<void> {
-    await this.changeToMember(ALL_GUARDIAN_INDEXES, guardianNumber, "guardians", page);
+  async zkFoundation(page?: Page) {
+    await this.switchToIndex(DERIVATION_INDEXES.ZK_FOUNDATION, page);
   }
 
-  async zkFoundation(page: Page): Promise<void> {
-    await this.switchToIndex(3, page);
+  async visitor(page?: Page) {
+    await this.switchToIndex(DERIVATION_INDEXES.VISITOR, page);
   }
 
-  async visitor(page: Page): Promise<void> {
-    await this.switchToIndex(4, page);
+  async randomCouncil(page?: Page) {
+    if (fastMode) {
+      return this.council(1, page);
+    }
+    return this.council(
+      faker.number.int({ min: 1, max: COUNCIL_SIZE }) as IntRange<1, typeof COUNCIL_SIZE>,
+      page
+    );
   }
 
-  private async switchToIndex(index: number, page: Page) {
-    let i = await this.countAccounts();
-    while (i < index) {
-      await this.wallet.createAccount();
-      i += 1;
+  async randomGuardian(page?: Page) {
+    if (fastMode) {
+      return this.guardian(1, page);
     }
-
-    await this.wallet.switchAccount(index);
-    await page.bringToFront();
+    return this.guardian(
+      faker.number.int({ min: 1, max: GUARDIANS_SIZE }) as IntRange<1, typeof GUARDIANS_SIZE>,
```

Comment:
This literally returns a random guardian, right? I don't love having randomness in tests. I like the tests that are in control of everything, in general. If this fails we don't know which council made it fail.

If we consider that it's important to test every member, we should a test for every member.

Reply:
Sounds good. WIll remove it.